### DISCUSSION TITLE
Saving and loading projects

### DISCUSSION THREAD

**Discussion by BerneS-1849361** [leon-thomm/Ryven#110] [Q&A]
Is there a way to save a project and open an other project from the code? And can i automaticly start ryven with a specific project open? Instead of having to select load project and open the project from there.

**Comment by leon-thomm** [comment]
**saving project**
- From inside a node? probably shouldn't be done (in case you need it for use with ryvencore you can use the API on the ryvencore session `self.session.serialize()`, something like that)
- From a parent python application that started Ryven from code? Currently would be `MainWindow.save_project()`, but this will  be exposed in a much more controlled manner, not from `MainWindow` (probably in next version)

**loading project**
- From inside a node? I don't think there's a nice way to do that.
- In case of running Ryven from code, the project will be indicated through parameters to an exposed `run()` function that launches the editor in the next version, but it's not supported to reload another project in the same editor session

**configure a project to be loaded automatically**
- The next version will feature command-line arguments which let you specify a project very quickly, and, as written above, when launching it from python you can just give the project as argument to the `run()` function

Does that answer your question?

**Reply by leon-thomm** [reply]
so, I think you were asking about the second case, running Ryven from code like
```python
import ryven
...
ryven.run_ryven(args)
```
then yes, saving project and loading specific projects will be supported in the next version
